src/notebooks/student_logins.ipynb

```diff
@@ -0,0 +1,277 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
+    "import os\n",
+    "import lms_utilities as lms\n",
+    "import matplotlib as plt"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# set the directory path for the output files from schoology extractor\n",
+    "csv_path = os.path.join(\"..\", \"schoology-extractor\", \"data\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# pull user data into a dataframe\n",
+    "users_df = lms.get_all_users(csv_path)\n",
+    "\n",
+    "# pull login data into a dataframe\n",
+    "logins_df = lms.get_all_system_activities(csv_path)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# merge user and login data for schoology\n",
+    "user_logins_df = pd.DataFrame()\n",
+    "user_logins_df = pd.merge(users_df, logins_df, how='inner', left_on=('SourceSystemIdentifier', 'SourceSystem'), right_on=('LMSUserSourceSystemIdentifier', 'SourceSystem'))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# add user and activity columns to login frequency dataframe\n",
+    "login_df = pd.DataFrame()\n",
+    "login_df = user_logins_df.loc[:,['Name', 'EmailAddress', 'SISUserIdentifier','ActivityType']].query(\"ActivityType not in ['sign-out']\")\n",
```

Comment:
You can simplify a little: since you want all rows, it is unnecessary to use `.loc`:
```suggestion
    "login_df = user_logins_df.[['Name', 'EmailAddress', 'SISUserIdentifier','ActivityType']].query(\"ActivityType not in ['sign-out']\")\n",
```

Reply:
i originally did not include the .loc but got a warning message recommending it.